src/protocol/sort/compose.rs

```diff
@@ -0,0 +1,127 @@
+use crate::{
+    error::BoxError,
+    ff::Field,
+    protocol::{context::ProtocolContext, reveal::reveal_a_permutation},
+    secret_sharing::Replicated,
+};
+use embed_doc_image::embed_doc_image;
+
+use super::{
+    apply::apply_inv,
+    shuffle::{get_two_of_three_random_permutations, Shuffle},
+    ComposeStep::{RevealPermutation, ShuffleSigma, UnshuffleRho},
+};
+
+/// This is an implementation of Compose (Algorithm 5) found in the paper:
+/// "An Efficient Secure Three-Party Sorting Protocol with an Honest Majority"
+/// by K. Chida, K. Hamada, D. Ikarashi, R. Kikuchi, N. Kiribuchi, and B. Pinkas
+/// <https://eprint.iacr.org/2019/695.pdf>
+/// This protocol composes two permutations by applying one secret-shared permutation(sigma) to another secret-shared permutation(rho)
+/// Input: First permutation(sigma) i.e. permutation that sorts all i-1th bits and other permutation(rho) i.e. sort permutation for ith bit
+/// Output: All helpers receive secret shares of permutation which sort inputs until ith bits.
+#[derive(Debug)]
+pub struct Compose<'a, F> {
+    sigma: &'a mut Vec<Replicated<F>>,
+    rho: &'a mut Vec<Replicated<F>>,
+}
+
+impl<'a, F: Field> Compose<'a, F> {
+    #[allow(dead_code)]
+    pub fn new(sigma: &'a mut Vec<Replicated<F>>, rho: &'a mut Vec<Replicated<F>>) -> Self {
+        Self { sigma, rho }
+    }
+    #[embed_doc_image("compose", "images/sort/compose.png")]
```

Comment:
I have a suggestion for a different diagram that I think would be more clear: https://docs.google.com/drawings/d/143VUS1-vs2j2SMwKluPMdQk7xco6yx-8JKEsWi36h9Q/edit?usp=sharing

Reply:
Thanks for the suggestion, using that instead. However, the template of this is different from other images. Once done with sort, will try to revisit all the diagrams to ensure correctness and consistent template